signer/src/transaction_coordinator.rs

```diff
@@ -1859,4 +1918,61 @@ mod tests {
     async fn should_ignore_withdrawals() {
         test_environment().assert_ignore_withdrawals().await;
     }
+
+    #[test_case(0, None, 1, 100, true; "first DKG allowed without min height")]
+    #[test_case(0, Some(100), 1, 5, true; "first DKG allowed regardless of min height")]
+    #[test_case(1, None, 2, 100, false; "subsequent DKG not allowed without min height")]
+    #[test_case(1, Some(101), 1, 100, false; "subsequent DKG not allowed with current height lower than min height")]
+    #[test_case(1, Some(100), 1, 100, false; "subsequent DKG not allowed when target rounds reached")]
+    #[test_case(1, Some(100), 2, 100, true; "subsequent DKG allowed when target rounds not reached and min height met")]
+    #[test_log::test(tokio::test)]
+    async fn test_should_coordinate_dkg(
+        dkg_rounds_current: u32,
+        dkg_min_bitcoin_block_height: Option<u64>,
+        dkg_target_rounds: u32,
+        chain_tip_height: u64,
+        should_allow: bool,
+    ) {
+        let context = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .modify_settings(|s| {
+                s.signer.dkg_min_bitcoin_block_height =
+                    dkg_min_bitcoin_block_height.map(NonZeroU64::new).flatten();
```

Comment:
nit: I think this can be simplified a little:
```rust
dkg_min_bitcoin_block_height.and_then(NonZeroU64::new);
```

Reply:
[680d822](https://github.com/stacks-network/sbtc/pull/1212/commits/680d82203c7c447d51ab8f84aa8d0b1685c533b5)